lib/opal/nodes/top.rb

```diff
@@ -108,6 +172,66 @@ def compile_end_construct
       def version_comment
         "/* Generated by Opal #{Opal::VERSION} */"
       end
+
+
+      def self.module_name(filenamepath, original_filepath = nil, original_filename = nil)
+        original_filename = filenamepath unless original_filename
+        original_filepath = Pathname.new(filenamepath).expand_path unless original_filepath
+        o_s = original_filepath.to_s
```

Comment:
Not sure if it's worth starting this method with `Pathname` since you use it mostly as a string. Maybe convert it to the string in the beginning and wrap it with a `Pathname` in the end?